pyomo/contrib/parmest/tests/test_parmest.py

```diff
@@ -26,17 +27,419 @@
+)
+@unittest.skipIf(not ipopt_available, "The 'ipopt' command is not available")
+
+# we use parameterized_class to test the two objective functions
+# over the two cases of the measurement error. Included a third objective function
+# to test the error message when an incorrect objective function is supplied
+@parameterized_class(
+    ("measurement_std", "objective_function"),
+    [
+        (None, "SSE"),
+        (None, "SSE_weighted"),
+        (None, "incorrect_obj"),
+        (0.1, "SSE"),
+        (0.1, "SSE_weighted"),
+        (0.1, "incorrect_obj"),
+    ],
+)
+class TestRooneyBieglerWSSE(unittest.TestCase):
+
+    def setUp(self):
+        from pyomo.contrib.parmest.examples.rooney_biegler.rooney_biegler import (
+            RooneyBieglerExperiment,
+        )
+
+        self.data = pd.DataFrame(
+            data=[[1, 8.3], [2, 10.3], [3, 19.0], [4, 16.0], [5, 15.6], [7, 19.8]],
+            columns=["hour", "y"],
+        )
+
+        # create the Experiment class
+        class RooneyBieglerExperimentWSSE(RooneyBieglerExperiment):
+            # def __init__(self, data, measurement_error_std):
+            #     self.data = data
+            #     self.model = None
+            #     self.measurement_error_std = measurement_error_std
+            def __init__(self, data, measurement_error_std):
+                super().__init__(data)
+                self.measurement_error_std = measurement_error_std
+
+            def label_model(self):
+                m = self.model
+
+                # add experiment outputs
+                m.experiment_outputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.experiment_outputs.update([(m.y[self.data['hour']], self.data['y'])])
+
+                # add unknown parameters
+                m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.unknown_parameters.update(
+                    (k, pyo.value(k)) for k in [m.asymptote, m.rate_constant]
+                )
+
+                # add measurement error
+                m.measurement_error = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.measurement_error.update(
+                    [(m.y[self.data['hour']], self.measurement_error_std)]
+                )
+
+                return m
+
+        # Create an experiment list
+        exp_list = []
+        for i in range(self.data.shape[0]):
+            exp_list.append(
+                RooneyBieglerExperimentWSSE(self.data.loc[i, :], self.measurement_std)
+            )
+
+        self.exp_list = exp_list
+
+        if self.objective_function == "incorrect_obj":
+            with pytest.raises(
+                ValueError,
+                match=r"Invalid objective function: 'incorrect_obj'\. "
+                r"Choose from: \['SSE', 'SSE_weighted'\]\.",
+            ):
+                self.pest = parmest.Estimator(
+                    self.exp_list, obj_function=self.objective_function, tee=True
+                )
```

Comment:
The basic parmest tests should already be testing for invalid input values to the `Estimator` constructor. I think you should rework this option/test to check the behavior when a user supplies a custom objective function. In that case you shouldn't see an error until you try to run `cov_est`.